Mathlib/LinearAlgebra/AffineSpace/Basis.lean

```diff
@@ -277,6 +275,24 @@ instance instAddAction : AddAction V (AffineBasis ι k P) :=
 
 end Ring
 
+section CommRing
+variable [CommRing k] [Module k V] (b : AffineBasis ι k P) {s : Finset ι} {i j : ι} (e : ι ≃ ι')
+
+instance instSMul : SMul kˣ (AffineBasis ι k V) where
+  smul a b :=
+    { toFun := a • ⇑b,
+      ind' := b.ind'.smul,
+      tot' := by
+        rw [Pi.smul_def, ← smul_set_range, Units.smul_def, ← AffineSubspace.smul_span, b.tot,
+          AffineSubspace.smul_top a.isUnit] }
+
+@[simp, norm_cast] lemma coe_smul (a : kˣ) (b : AffineBasis ι k V) : ⇑(a • b) = a • ⇑b := rfl
```

Comment:
Can you add some lemmas about the other basic operations too, like we did for vadd?

Reply:
The ones I don't have here are the ones I failed to prove. Feel free to add them yourself though